### ISSUE TITLE
git sync: fails when it is run in a new directory

### ISSUE THREAD

**Issue by retospect** [Originate/git-town#371]
When I add a new directory and commit a new file within, then run sync, I get dumped back into the master branch, and the system remains in a weird state (namely, the current directory does no longer exist). 

I'd like it to handle the new directory gracefully, as it does if I only add a new file. 

```
OL ~/Documents/sb/rms/asf-deploy>master>$ git hack branchy
...
OL ~/Documents/sb/rms/asf-deploy>branchy>$ mkdir newDir
OL ~/Documents/sb/rms/asf-deploy>branchy>$ cd newDir
OL ~/Documents/sb/rms/asf-deploy/newDir>branchy>$ echo Hello > wibble
OL ~/Documents/sb/rms/asf-deploy/newDir>branchy*>$ git add wibble
OL ~/Documents/sb/rms/asf-deploy/newDir>branchy*>$ git commit -m "Wibble"
...
OL ~/Documents/sb/rms/asf-deploy/newDir>branchy>$ git sync

[branchy] git fetch --prune

[branchy] git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory

[] git push -u origin 
fatal: Unable to read current working directory: No such file or directory

To abort, run "git sync --abort".
To continue after you have resolved the conflicts, run "git sync --continue".
fatal: Unable to read current working directory: No such file or directory
fatal: Unable to read current working directory: No such file or directory
To skip the sync of the '' branch, run "git sync --skip".
```

This needs to be added to the following commands:
- [ ] git extract
- [ ] git hack
- [ ] git ship
- [ ] git sync
- [ ] git sync-fork

**Comment by kevgo** [comment]
+1 it should skip the remote operations

**Comment by kevgo** [comment]
This is a weird situation. Git happily changes to the `master` branch, in which that new "newDir" directory in which you currently are does not exist. Git doesn't notice that, and emits no error code. This leaves the current shell in limbo, thereby being unable to do any operations (`ls`, `mkdir`, `git` etc). The only thing that works is to `cd` into another directory.

Ultimately I think this is a bug in Git. Given that the user is left in limbo, Git should reject this checkout.  However, to address this, and to enforce the best practice to run Git commands always from the root directory of the repo, we could 
* remember the current working directory
* `cd` into the root directory of the Git repo (`cd "$(git rev-parse --show-toplevel)"`
* run the `git sync` stuff there
* afterwards `cd` back into the stored directory

This sounds like the right way to use Git anyways, so let's do this.

**Comment by kevgo** [comment]
Implemented in #458
